jazz_environments/index.js

```diff
@@ -27,189 +27,204 @@ const logger = require("./components/logger.js"); //Import the logging module.
+                domain = event.query.domain.toLowerCase();
+
+                var update_environment_payload = Object.assign({}, event.body);
+
+                validateUpdateInput(update_environment_payload, environment_id)
+                .then(() => validateEnvironmentExists(env_tableName, indexName, service, domain, environment_id))
+                .then((result) => updateServiceEnvironment(env_tableName, update_environment_payload, result.data))
+                .then(function(result){
+                    logger.info("Environment update success:"+JSON.stringify(result));
+                    return cb(null, responseObj(result, update_environment_payload));
+                })
+                .catch(function(err){
+                    logger.error("Error while updating environment catalog:"+JSON.stringify(err));
+                    if (err.errorType) {
+                        // error has already been handled and processed for API gateway
+                        return cb(JSON.stringify(err));
+                    }else {
+                        if (err.result === "inputError") {
+                            return cb(JSON.stringify(errorHandler.throwInputValidationError(err.message)));
+                        }
+
+                        return cb(JSON.stringify(errorHandler.throwInternalServerError("Unexpected Error occured.")));
+                    }
+                });
+            }
+
+            // Create new service environment
+            // 6: POST a service
+            if (event.method === "POST" && event.body) {
+                var environment_data = event.body;
+                logger.info("Create new environment with the following data:" + JSON.stringify(environment_data));
+
+                validateEnvironmentData(env_tableName, environment_data, indexName)
+                .then(() => addNewEnvironment(environment_data, env_tableName))
+                .then(function(result){
+                    logger.info("New environment created:"+JSON.stringify(result));
+                    return cb(null, responseObj(result, environment_data));
+                })
+                .catch(function(err){
+                    logger.error("error while creating new environment:"+JSON.stringify(err));
+                    if (err.errorType) {
+                        // error has already been handled and processed for API gateway
+                        return cb(JSON.stringify(err));
+                    }else {
+                        if (err.result === "inputError") {
+                            return cb(JSON.stringify(errorHandler.throwInputValidationError(err.message)));
+                        }
+
+                        return cb(JSON.stringify(errorHandler.throwInternalServerError("Unexpected Error occured.")));
+                    }
+                });
+            }
+        })
+        .catch(function(error){
+            logger.debug("Invalid event params");
```

Comment:
This debug message is misleading - we don't know if this is for invalid event params, yet.

Reply:
removed